File: world/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void) {
    int change;

    // Prompt the user for the amount of change owed
    do {
        change = get_int("Change owed: ");
    } while (change < 0);

    // Define the coin values
    int coins[] = {25, 10, 5, 1};
    int num_coins = 0;

    // Calculate the minimum number of coins
    for (int i = 0; i < 4; i++) {
        num_coins += change / coins[i];
        change %= coins[i];
    }

    printf("%d\n", num_coins);

}

File: sentimental-cash/cash.py
def get_float(prompt):
    while True:
        try:
            value = float(input(prompt))
            if value >= 0:
             return value
        except ValueError:
            print("Invalid input. Please enter a non-negative number.")
            continue

def calculate_coins():
    change = get_float("Enter the amount of change owed: ")

    cents = round(change * 100)
    quarters = 25
    dimes = 10
    nickels = 5
    pennies = 1

    num_coins = 0
    num_coins += cents // quarters
    cents %= quarters
    num_coins += cents // dimes
    cents %= dimes
    num_coins += cents // nickels
    cents %= nickels
    num_coins += cents // pennies

    print(num_coins)
calculate_coins()

